.github/workflows/pr-testing-with-test-project.yml

```
@@ -1,13 +1,37 @@
-#This workflow runs the tests in the test projects to make sure the generator works as a library where it is a Node dependency along with the template.
 name: Test using test project
 
 on:
   pull_request:
     types: [opened, reopened, synchronize, ready_for_review]
+    paths:
+      - 'package.json'
+      - 'package-lock.json'
+      - 'apps/'
+      - 'packages/'
+      - '.github/workflows/'
+      - '!/*.md'
+      - '!docs/**'
 
 jobs:
+  should-test:
+    runs-on: ubuntu-latest
+    outputs:
+      run_tests: ${{ steps.check.outputs.run_tests }}
+    steps:
+      - name: Check if tests should run
+        id: check
+        run: |
+          # Skip for bot PRs with specific titles
+          if [[ "${{ github.actor }}" == "asyncapi-bot" && "${{ github.event.pull_request.title }}" =~ ^(ci:\ update|chore\(release\):) ]] || \
+             [[ "${{ github.actor }}" == "allcontributors[bot]" && "${{ github.event.pull_request.title }}" =~ ^docs: ]]; then
+            echo "run_tests=false" >> $GITHUB_OUTPUT
+          else
+            echo "run_tests=true" >> $GITHUB_OUTPUT
+          fi
+
   test:
-    if: github.event.pull_request.draft == false
+    needs: should-test
+    if: needs.should-test.outputs.run_tests == 'true' && github.event.pull_request.draft == false
```

```suggestion
    if: needs.should-test.outputs.run_tests == true && github.event.pull_request.draft == false
```

wrapping `true` in single quotes makes it string, although eventually, it results in a true statement (as strings are also considered to be true) so the workflow works as expected. But it's a good practice to use Booleans for truthy and falsy statements.

---

Alright got it !! i took the assumptions so did such changes but anyway will resolve it .